MetaMask/metamask-extension
# fix: token rates controller start / stop logic

## **Description**

Problem: Users reporting missing fiat prices for some erc20 tokens, even when price setting is on:

<img width="357" alt="Screenshot 2024-03-14 at 2 35 41 PM" src="https://github.com/MetaMask/metamask-extension/assets/3500406/711cfa40-7387-43df-9981-9ed52973b5b4">

Diagnosis:

(1) The `TokenRatesController` gets stopped when metamask is closed.  But it is not started again when metamask re-opens, preventing any future token price updates.

(2) If you have metamask installed but you never open/unlock it (within a browser session), the `TokenRatesController` is still polling in the background every 3 minutes.

The fix for both is to start the controller in `triggerNetworkrequests` instead of on construction.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24021?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Restore a wallet where multiple accounts have erc20 tokens
2. Enable token autodetection
4. Visit the first account with tokens, verify their prices
5. Close metamask
6. Re-open metamask
7. Visit the second account
8. Verify its tokens have prices

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
